### PR TITLE
add chainlit sample 

### LINKED ISSUE #4748: Add Sample on how to use AgentChat with app libraries like Chainlit/Streamlit
Add examples on how to use the new AutoGen v0.4 api  with UI libraries like [chainlit](https://github.com/Chainlit/chainlit).

### PR DESCRIPTION
**Pull request by victordibia**
Adds a sample on how to use Chainlit with AgentChat 

image

## Related issue number

Closes #4748

Future PR might extend this to show how to do something similar with Core.

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by husseinmozannar** [comment]
Going through it:

so far got this error

    autogen-core 0.4.0.dev11 depends on protobuf~=4.25.1
    opentelemetry-proto 1.29.0 depends on protobuf<6.0 and >=5.0
    
    ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts

**Review by ekzhu — APPROVED** [review]
We can merge for now and fix the protobuf dependency issue in another PR.

**Comment by ekzhu** [comment]
> Going through it:
> 
> so far got this error
> 
> ```
> autogen-core 0.4.0.dev11 depends on protobuf~=4.25.1
> opentelemetry-proto 1.29.0 depends on protobuf<6.0 and >=5.0
> 
> ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
> ```

Can you create an issue for this. Simply installing `chainlit` with `autogen-core`

**Review by husseinmozannar — CHANGES_REQUESTED** [review]
- need to reset termination
- add sample for userproxy 

(im doing it now)

**Review by husseinmozannar — APPROVED** [review]

**Comment by husseinmozannar** [comment]
My changes:
- put the team/agent/termination creation code inside run_team so that you can chat for many times like in AGS 
- add a message to signal that the team work is done "termination condition met"
- add to readme how to add userproxyagent (it actually works neatly!)
- make it run in headless - the pop up browser is bad

**Review by gagb — COMMENTED** [review]
I am getting this error:

```console
Traceback (most recent call last):
  File "/Users/gaganbansal/workspace/autogen/python/samples/agentchat_chainlit/.venv/bin/chainlit", line 5, in <module>
    from chainlit.cli import cli
  File "/Users/gaganbansal/workspace/autogen/python/samples/agentchat_chainlit/.venv/lib/python3.11/site-packages/chainlit/__init__.py", line 21, in <module>
    from chainlit.action import Action
  File "/Users/gaganbansal/workspace/autogen/python/samples/agentchat_chainlit/.venv/lib/python3.11/site-packages/chainlit/action.py", line 5, in <module>
    from chainlit.telemetry import trace_event
  File "/Users/gaganbansal/workspace/autogen/python/samples/agentchat_chainlit/.venv/lib/python3.11/site-packages/chainlit/telemetry.py", line 12, in <module>
    from chainlit.config import config
  File "/Users/gaganbansal/workspace/autogen/python/samples/agentchat_chainlit/.venv/lib/python3.11/site-packages/chainlit/config.py", line 532, in <module>
    config = load_config()
             ^^^^^^^^^^^^^
  File "/Users/gaganbansal/workspace/autogen/python/samples/agentchat_chainlit/.venv/lib/python3.11/site-packages/chainlit/config.py", line 501, in load_config
    settings = load_settings()
               ^^^^^^^^^^^^^^^
  File "/Users/gaganbansal/workspace/autogen/python/samples/agentchat_chainlit/.venv/lib/python3.11/site-packages/chainlit/config.py", line 473, in load_settings
    code_settings = CodeSettings(action_callbacks={})
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/gaganbansal/workspace/autogen/python/samples/agentchat_chainlit/.venv/lib/python3.11/site-packages/pydantic/_internal/_dataclasses.py", line 121, in __init__
    s.__pydantic_validator__.validate_python(ArgsKwargs(args, kwargs), self_instance=s)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/gaganbansal/workspace/autogen/python/samples/agentchat_chainlit/.venv/lib/python3.11/site-packages/pydantic/_internal/_mock_val_ser.py", line 100, in __getattr__
    raise PydanticUserError(self._error_message, code=self._code)
pydantic.errors.PydanticUserError: `CodeSettings` is not fully defined; you should define `Action`, then call `pydantic.dataclasses.rebuild_dataclass(CodeSettings)`.

For further information visit https://errors.pydantic.dev/2.10/u/class-not-fully-defined
```

**Comment by victordibia** [comment]
@gagb , what is your chainlit version?
Just to confirm and help debug  
Anyway, it seems this is a chainlit issue that is being fixed in the next release - (https://github.com/Chainlit/chainlit/issues/1544#issuecomment-2541496286)

**Comment by gagb** [comment]
> @gagb , what is your chainlit version? Just to confirm and help debug Anyway, it seems this is a chainlit issue that is being fixed in the next release - ([Chainlit/chainlit#1544 (comment)](https://github.com/Chainlit/chainlit/issues/1544#issuecomment-2541496286))

chainlit==2.0rc0

File: python/samples/agentchat_chainlit/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+- `start_chat`: Initializes the chat session and sets up the avatar for Claude.
+- `run_team`: Sends the user's query to the Anthropic API and streams the response back to the chat interface.
+- `chat`: Receives messages from the user and passes them to the `call_claude` function.
+
+## Next Steps (Extra Credit)
+
+In this example, we created a basic AutoGen team with a single agent, Claude. There are a few ways you can extend this example:
+
+- Add more agents to the team.
+- Explor custom agents that sent multimodal messages
```
Comment (benoitf73) [inline python/samples/agentchat_chainlit/README.md:107]: Small Typo:
- Explore custom agents that sent multimodal messages

**Pull request merged**